From ee5cd8bfa6fc9a11dd1048e3e086f63c2b7a0b37 Mon Sep 17 00:00:00 2001
From: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>
Date: Mon, 2 Sep 2024 18:06:25 -0400
Subject: [PATCH] thunderbird: deprecate darwinSetupWarning

Now that `profiles.ini` doesn't list `Version=2`, setting the
environment variables is no longer needed: implicit version is v1, which
means `[Install]` sections are stored in a separate `installs.ini` file
that is managed by Thunderbird itself, which `profiles.ini` stays
read-only.

Signed-off-by: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>
---
 modules/programs/thunderbird.nix | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/modules/programs/thunderbird.nix b/modules/programs/thunderbird.nix
index 41663f152d27..ce0894ec9730 100644
--- a/modules/programs/thunderbird.nix
+++ b/modules/programs/thunderbird.nix
@@ -231,7 +231,7 @@ in {
 
       darwinSetupWarning = mkOption {
         type = types.bool;
-        default = true;
+        default = false;
         example = false;
         visible = isDarwin;
         readOnly = !isDarwin;
@@ -324,17 +324,8 @@ in {
     ];
 
     warnings = optional (isDarwin && cfg.darwinSetupWarning) ''
-      Thunderbird packages are not yet supported on Darwin. You can still use
-      this module to manage your accounts and profiles by setting
-      'programs.thunderbird.package' to a dummy value, for example using
-      'pkgs.runCommand'.
-
-      Note that this module requires you to set the following environment
-      variables when using an installation of Thunderbird that is not provided
-      by Nix:
-
-          export MOZ_LEGACY_PROFILES=1
-          export MOZ_ALLOW_DOWNGRADE=1
+      Using programs.thunderbird.darwinSetupWarning is deprecated. The module
+      is now compatible with all thunderbird installations.
     '';
 
     home.packages = [ cfg.package ]
